sympy/integrals/prde.py

```diff
@@ -918,15 +919,46 @@ def parametric_log_deriv_heu(fa, fd, wa, wd, DE, c1=None):
 
 
 def parametric_log_deriv(fa, fd, wa, wd, DE):
-    # TODO: Write the full algorithm using the structure theorems.
-#    try:
-    A = parametric_log_deriv_heu(fa, fd, wa, wd, DE)
-#    except NotImplementedError:
-        # Heuristic failed, we have to use the full method.
-        # TODO: This could be implemented more efficiently.
-        # It isn't too worrisome, because the heuristic handles most difficult
-        # cases.
-    return A
+    """
+    >>> fa, fd = Poly(5*t**2 + t - 6, t), Poly(2*x*t**2, t)
+    >>> wa, wd = Poly(-1, t), Poly(x*t**2, t)
+    >>> DE = DifferentialExtension(extension={'D': [Poly(1, x), Poly(1/x, t)],
+        'exts': [None, 'log']})
+    >>> parametric_log_deriv(fa, fd, wa, wd, DE)
+    (Matrix([
+    [1, 0],
+    [0, 1]
+    ]), Matrix([
+    [-(10*t**4*x + 2*t**3*x - 10*t**2*x - 24*t*x)/(10*t + 1)],
+    [(2*t**2*x + 4*t*x)/(10*t + 1)]]))
+
+    Page, 251 section 7.3
+
+    """
+
+    F = []
+    for i, ext in enumerate(DE.exts):
+        if ext == 'exp':
+            F.append(DE.D[i]/DE.T[i])
+        elif ext == 'log':
+            F.append(DE.D[i].as_expr())
+
+    # find a field containing all the fractions
+    FF, gens = sfield(F, extension=True)
+
+    denom = gens[0].denom
+    for gen in gens[1: ]:
+        denom = denom.lcm(gen.denom)
+
+    nums = [denom.exquo(f.denom)*f.numer for f in gens]
+    monoms = set()
+    for num in nums:
+        monoms.update(num.monoms())
+
+    # each monomial corresponds to a row in the matrix
+    # of entries in 'K' giving the relations.
+    # The elements of F correspond to the columns
+    t = [[num._get_coeff(m) for num in nums] for m in monoms]
```

Comment:
The matrix will be needed to find the nullspace after the entries are made rational.